src/src/pages/capabilities/KafkaCluster/MessageContracts.js

```diff
@@ -0,0 +1,151 @@
+import styles from "./MessageContract.module.css";
+import Expandable from "../../../components/Expandable";
+import Poles from "../../../components/Poles";
+import { Text } from "@dfds-ui/typography";
+import { SelectField, Switch } from "@dfds-ui/react-components";
+import React, { useEffect, useState } from "react";
+import { MessageHeader, MessageStatus } from "./MessageContract";
+import { Divider } from "@dfds-ui/react-components/divider";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import { vs as syntaxStyle } from "react-syntax-highlighter/dist/esm/styles/hljs";
+
+function JsonViewer({ json }) {
+  return (
+    <div className={styles.messagecontent}>
+      <SyntaxHighlighter
+        language="json"
+        style={syntaxStyle}
+        wrapLongLines={false}
+        customStyle={{
+          margin: "0",
+          padding: "0",
+          border: "1px solid #ccc",
+          height: "370px",
+        }}
+      >
+        {json}
+      </SyntaxHighlighter>
+    </div>
+  );
+}
+
+export default function MessageContracts({
+  id,
+  messageType,
+  contracts,
+  onRetryClicked,
+  isSelected,
+  onHeaderClicked,
+}) {
+  const [canExpand, setCanExpand] = useState(false);
+  const [headerStatus, setHeaderStatus] = useState(MessageStatus.PROVISIONED);
+  const [selectedContract, setSelectedContract] = useState([]);
+  const [shownContracts, setShownContracts] = useState([]);
+  const [showSchema, setShowSchema] = useState(false);
+  const handleToggleSchema = () => {
+    setShowSchema((prev) => !prev);
+  };
+
+  useEffect(() => {
+    if (contracts) {
+      setShownContracts(contracts);
+      if (contracts.length === 1) {
+        setCanExpand(contracts[0].status === MessageStatus.PROVISIONED);
+        setHeaderStatus(contracts[0].status);
+      } else if (contracts.length > 1) {
+        setCanExpand(true);
+      }
+    }
+  }, [contracts]);
+
+  const headerClickHandler = () => {
+    if (onHeaderClicked) {
+      onHeaderClicked(messageType);
+      setSelectedContract(contracts.find((x) => x.Id === id));
```

Comment:
id is not set, maybe we want to just save the ID of the first version.

Reply:
are we 100% sure that the first version will be always saved as SchemaVersion = 1